qml/TimeZonePage.qml

```diff
@@ -210,28 +269,110 @@ BasePage {
 
         }
     }
 
+
+
     Column {
         id: column
         anchors.fill: content
-        spacing: Units.gu(5)
+        spacing: Units.gu(2)
+
+        Row {
+            id: timeFormatRow
+            width: parent.width
+            Column{
+                anchors.verticalCenter: timeFormatRow.verticalCenter
+                width: parent.width - Units.gu(8)
+                Text {
+                    id: timeFormatText
+                    text: "Time Format"
+                    font.pixelSize: Units.gu(36/13.5)
+                    color: "white"
+                    anchors.left: parent.left
+                    width: parent.width - Units.gu(8)
+                }
+            }
+            Column
+            {
+                anchors.verticalCenter: timeFormatRow.verticalCenter
+                Switch
+                {
+                    id: timeFormatSwitch
+                    anchors.right: parent.right
+                    checked: currentTimeFormat === "HH12" ? false : true
+                    style: SwitchStyle {
+                        groove: Image
+                        {
+                        id: grooveImage
+                        source: timeFormatSwitch.checked ? "images/toggle-button-on.png" : "images/toggle-button-off.png"
+                        width: Units.gu(8)
+                        height: Units.gu(4)
+
+                        Text
+                        {
+                            color: "white"
+                            text: "24H"
+                            font.bold: true
+                            font.family: "Prelude"
+                            font.pixelSize: FontUtils.sizeToPixels("small")
+                            visible: timeFormatSwitch.checked
+                            anchors.verticalCenter: parent.verticalCenter
+                            anchors.left: parent.left
+                            anchors.leftMargin: Units.gu(1)
+                        }
+                        Text
+                        {
+                            color: "white"
+                            text: "12H"
+                            font.bold: true
+                            font.family: "Prelude"
+                            font.pixelSize: FontUtils.sizeToPixels("small")
+                            visible: !timeFormatSwitch.checked
+                            anchors.verticalCenter: parent.verticalCenter
+                            anchors.right: parent.right
+                            anchors.rightMargin: Units.gu(1)
+                        }
+
+                    }
+                    handle: Rectangle {
+                        color: "transparent"
+                    }
+                }
+                onClicked:
+                {
+                    timeFormat = timeFormatSwitch.checked ? "HH24" : "HH12"
+                    service.call("luna://com.palm.systemservice/getPreferenceValues", JSON.stringify({
```

Comment:
This is a bit nasty and might be high on resources since we basically recreate the full listmodel.

I tried a few other things like listmodel.setProperty, but I could never get the time to switch between 12H and 24H format on the fly. I'm open for suggestions. Basically the time displayed in the timezones should adjust automatically when the toggle is switched. This works, but it's not very elegant.